orttraining/orttraining/python/training/ortmodule/torch_cpp_extensions/__init__.py

```diff
@@ -0,0 +1,45 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+"""Support for PyTorch C++ extensions within ORTModule
+
+
+TODO: Implement mechanism to register extensions and prevent issues with incorrect/missing flags
+      for each :meth:`torch.utils.cpp_extension.*` call
+"""
+
+import threading
+from functools import wraps
+from onnxruntime.capi import _pybind_state as C
+
+
+def run_once_aten_op_executor(f):
+    """
+    Decorator to run a function only once.
+    :param f: function to be run only once during execution time despite the number of calls
+    :return: The original function with the params passed to it if it hasn't already been run before
+    """
+    @wraps(f)
+    def aten_op_executor_wrapper(*args, **kwargs):
+        if not aten_op_executor_wrapper.has_run:
```

Comment:
Why check twice (before and after lock)?

Reply:
This code is unchanged from @iK1D